src/components/ImageView/index.native.js

```diff
@@ -186,16 +166,24 @@ class ImageView extends PureComponent {
                         this.imageZoomScale = scale;
                     }}
                 >
-                    <Image
+                    <FastImage
                         style={[
                             styles.w100,
                             styles.h100,
                             this.props.style,
+
+                            // Hide image while loading so ImageZoom can get the image
+                            // size before presenting - preventing visual glitches or shift
+                            // due to ImageZoom
+                            this.state.isLoading ? styles.opacity0 : styles.opacity1,
                         ]}
-                        source={{uri: this.props.url}}
-                        resizeMode="contain"
+                        source={{
+                            uri: this.props.url,
+                            headers: this.props.isAuthTokenRequired ? chatAttachmentTokenHeaders() : undefined,
```

Comment:
Do you think it makes sense to move this to the beginning of the `render` function, like we have elsewhere? a.k.a. calculating `const headers = this.props.isAuthTokenRequired ? chatAttachmentTokenHeaders() : undefined;` at the beginning, for consistency?

Reply:
Yes sounds good - I'll move it :)